model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1559,6 +1559,11 @@ message StandardProtocols {
     // cores.
     MULTI_CORE_BUNDLE_PROCESSING = 3 [(beam_urn) = "beam:protocol:multi_core_bundle_processing:v1"];
 
+    // Indicates this SDK can cheaply spawn sibling workers (e.g. within the
+    // same container) to work around the fact that it cannot take advantage
+    // of multiple cores (i.e. MULTI_CORE_BUNDLE_PROCESSING is not set).
+    SIBLING_WORKERS = 5 [(beam_urn) = "beam:protocol:sibling_workers:v1"];
```

Comment:
Do you think we could get python to finally support multi core bundle processing by using sub-interpreters / PEP554 instead of sibling workers or is the overhead not practical because we would need to still pay a high cost to transfer objects between the main interpreter that would have the connections to the runner and the sub-interpreters that would do the work?

Reply:
Yes, sub-interpreters sharing the same connection to the runner would have many of the same downsides of a multi-process worker in that it's expensive to transfer objects (though maybe slightly cheaper than between processes, and some simple (immutable, leaf) types can be shared. If one can't easily share objects, there's little benefit to using sub-interpreters over independent processes. (Something to possibly keep an eye on though.)